theano/tensor/nnet/abstract_conv.py

```diff
@@ -1517,18 +1525,23 @@ def conv(self, img, kern, mode="valid", dilation=1):
                      ] = kern
 
         if self.convdim == 2:
+            if img.shape[1] % self.num_groups != 0:
+                raise ValueError("Number of input channels must be divisible by num_groups")
             val = _valfrommode(mode)
             bval = _bvalfromboundary('fill')
+            input_channel_offset = img.shape[1] // self.num_groups
+            output_channel_offset = kern.shape[0]
```

Comment:
After working out the finer details, I think the second method is better to implement for CorrMM and GPUCorrMM as we can easily divide the weight tensor into separate groups . 

Since this PR is a mess, should i close this one and open a new one  or make changes in this one itself?